### PR TITLE
Add support for kind cluster creation in the CLI

### LINKED ISSUE #108: Refactor common functionality for tiller and helm
The functions for creating clusters on different cloud providers all use similar code for deploying tiller and MLbench using pyhelm. We should refactor the code to pull out the common functionality in one function.

### LINKED ISSUE #253: Refactor CLI
There is a lot of code duplication in the CLI script. Also, the file is almost 2000 lines long so we could separate it into multiple chunks

### PR DESCRIPTION
**Pull request by mmilenkoski**
This PR adds support for creating kind clusters through the CLI. 

To create a cluster use: 

`mlbench create-cluster kind 3 my-cluster "/local/path/to/mlbench-helm"`

To delete the cluster use:

`mlbench delete-cluster kind my-cluster-3`

The other commands are the same as with gcloud. 

Closes #108 
Closes #253

### CODE REVIEW

File: mlbench_core/cli/cli.py [hunk 1]
```diff
@@ -391,6 +490,26 @@ def delete_gcloud(name, zone, project):
     click.echo("Cluster deleted.")
 
 
+@delete_cluster.command("kind")
+@click.option("--cluster_name", "-n", default="kind", type=str)
+def delete_gcloud(cluster_name):
```
Comment (cgrigis) [inline mlbench_core/cli/cli.py:495]: `delete_kind`?

File: mlbench_core/cli/cli.py [hunk 2]
```diff
@@ -803,110 +726,85 @@ def create_gcloud(
...
         # create local registry
         click.echo("Creating registry {}".format((registry_name)))
+        docker_client.containers.run(
+            image="registry:2",
+            name=registry_name,
+            restart_policy={"Name": "always"},
+            ports={registry_port: host_port},
+            detach=True,
+        )
+        while docker_client.containers.get(registry_name).status != "running":
```
Comment (Panaetius) [inline mlbench_core/cli/cli.py:765]: If there is something wrong and the container immediately terminates, the status could be `exited` and this loop would hang indefinitely.

Also it's good practice to add a sleep of 0.5-1 second here so the code is not busy waiting and using up unnecessary CPU

**Comment by ehoelzl** [comment]
@mmilenkoski Update on this PR ?

**Comment by mmilenkoski** [comment]
@ehoelzl I resolved the conflicts and it is ready for merge. Please approve it if you don't have any remarks.

**Comment by ehoelzl** [comment]
> @ehoelzl I resolved the conflicts and it is ready for merge. Please approve it if you don't have any remarks.

Have you tried using [pytest-kind](https://pypi.org/project/pytest-kind/) ? Can easily deploy clusters with that

**Comment by mmilenkoski** [comment]
> > @ehoelzl I resolved the conflicts and it is ready for merge. Please approve it if you don't have any remarks.
> 
> Have you tried using [pytest-kind](https://pypi.org/project/pytest-kind/) ? Can easily deploy clusters with that

I think it is used as a plugin for pytest, but I saw now that they say it can be used independently of pytest. I will check that out.

**Comment by mmilenkoski** [comment]
> > > @ehoelzl I resolved the conflicts and it is ready for merge. Please approve it if you don't have any remarks.
> > 
> > 
> > Have you tried using [pytest-kind](https://pypi.org/project/pytest-kind/) ? Can easily deploy clusters with that
> 
> I think it is used as a plugin for pytest, but I saw now that they say it can be used independently of pytest. I will check that out.

Hey @ehoelzl, I was looking into pytest-kind and they use exactly the same way with `subprocess` calls to create the clusters. I am not sure whether it is better to introduce the dependency or leave our own implementation. What do you think?

You can find their implementation [here](https://github.com/hjacobs/pytest-kind/blob/481e7c9f3a7d0a62f31a8f45a83fa5bc96420bf9/pytest_kind/cluster.py#L66)

**Comment by ehoelzl** [comment]
> Hey @ehoelzl, I was looking into pytest-kind and they use exactly the same way with `subprocess` calls to create the clusters. I am not sure whether it is better to introduce the dependency or leave our own implementation. What do you think?
> 
> You can find their implementation [here](https://github.com/hjacobs/pytest-kind/blob/481e7c9f3a7d0a62f31a8f45a83fa5bc96420bf9/pytest_kind/cluster.py#L66)

Only advantage I see is reduction of code base on our side, especially for a tool we use. Also, they have some functions that download kind and install it easily on kubectl to make it easier on the user (see [`ensure_kind`](https://github.com/hjacobs/pytest-kind/blob/481e7c9f3a7d0a62f31a8f45a83fa5bc96420bf9/pytest_kind/cluster.py#L30) and [`ensure_kubectl`](https://github.com/hjacobs/pytest-kind/blob/481e7c9f3a7d0a62f31a8f45a83fa5bc96420bf9/pytest_kind/cluster.py#L48))

**Review by ehoelzl — APPROVED** [review]

**Pull request merged**